### ISSUE TITLE
Permanent logout due to invalid session token

### ISSUE THREAD

**Issue by gillima** [phpList/phplist3#261]
I update from PHP5 to PHP7.1 on a running phplist installation. After the update i've got logged-out on every click after a successful login with the reason "invalid session token".

I tested with PHP sessions and session tables as well, the behavior was still the same.
Finally I found out that adding a `@session_start()` at the beginning of `admin/index.php` fixes the issue.

Environment: phplist 3.3.1, php 7.1.11, apache 2.4.26

**Comment by samtuke** [comment]
Sounds like a great catch, thanks!

Can someone else replicate this, @bramley @michield ?

**Comment by bramley** [comment]
My shared hosting site runs fine with phplist 3.3.1, php 7.1.14 and apache 2.4.29, so I guess that there is some other factor in play.

**Comment by gillima** [comment]
My PHP7 installations have `session.auto_start` not set, Sadly I cannot check how it was set before, but if it was set this would definitively prevent the issue...

**Comment by gillima** [comment]
Finally i found it. Seems not to be a problem in the phplist core! Sorry about that. I moved the session_start downwards until i found the line which caused the problem. Finally I ended inside config.php. Right after setting the database credentials there was a `init_set('session.save_path', ...)` to a non writable directory. I guess the previous version of PHP was just not allowed to set this path, that's why it worked...

Now it works even without the session_start because this is now done somehow later when including the languages.php file. Anyway, thanks for the help!!!
